### ISSUE TITLE
PDO Metadata Source Doesn't Load __DYNAMIC:1__ Metadata Sets

### ISSUE THREAD

**Issue by tdiscuit** [simplesamlphp/simplesamlphp#926]
It looks like version 1.16.x will not load any of the __DYNAMIC:1__ metadata sets with the PDO Metadata Storage Handler (I haven't tested it with the other types, but I believe those are unaffected as they don't override the `getMetaData` function from the `MetadataStorageSource` class).

On `frontpage_federation.php` it shows the metadata set with an Entity ID of `__DYNAMIC:1__` and Index of `__DYNAMIC:1__`. If you click on show metadata, it returns:

```
Backtrace:
0 www/saml2/idp/metadata.php:237 (N/A)
Caused by: SimpleSAML_Error_MetadataNotFound: METADATANOTFOUND('%ENTITYID%' => '\'https://idp.example.com/saml2/idp/metadata.php\'')
Backtrace:
2 lib/SimpleSAML/Metadata/MetaDataStorageHandler.php:299 (SimpleSAML_Metadata_MetaDataStorageHandler::getMetaData)
1 lib/SimpleSAML/Metadata/MetaDataStorageHandler.php:319 (SimpleSAML_Metadata_MetaDataStorageHandler::getMetaDataConfig)
0 www/saml2/idp/metadata.php:28 (N/A)
```

On `frontpage_federation.php` page in 1.15.4, it lists the IdP metadata with an Entity ID of `https://idp.example.com/saml2/idp/metadata.php` and an Index of `__DYNAMIC:1__`. Viewing the metadata set will properly show the metadata set as expected.

**Comment by thijskh** [comment]
@manu0401 Is this something you could look into?

**Comment by ct27stf** [comment]
so when will it be released for the 1.17 ?
